packages/generator-single-spa/src/svelte/generator-single-spa-svelte.js

```diff
@@ -0,0 +1,178 @@
+  constructor(args, opts) {
+    super(args, opts);
+
+    this.option("packageManager", {
+      type: String,
+    });
+    // this.option("typescript", {
+    //   type: Boolean,
+    // });
+    this.option("orgName", {
+      type: String,
+    });
+    this.option("projectName", {
+      type: String,
+    });
+  }
+  async createPackageJson() {
+    if (!this.options.packageManager) {
+      this.options.packageManager = (
+        await this.prompt([
+          {
+            type: "list",
+            name: "packageManager",
+            message: "Which package manager do you want to use?",
+            choices: ["yarn", "npm"],
+          },
+        ])
+      ).packageManager;
+    }
+
+    // if (this.options.typescript === undefined) {
+    //   this.options.typescript = (
+    //     await this.prompt([
+    //       {
+    //         type: "confirm",
+    //         name: "typescript",
+    //         message: "Will this project use Typescript?",
+    //         default: false,
+    //       },
+    //     ])
+    //   ).typescript;
+    // }
+
+    if (!this.options.orgName) {
+      this.options.orgName = (
+        await this.prompt([
+          {
+            type: "input",
+            name: "orgName",
+            message: "Organization name (use lowercase and dashes)",
+          },
+        ])
+      ).orgName;
+    }
+
+    if (!this.options.projectName) {
+      this.options.projectName = (
+        await this.prompt([
+          {
+            type: "input",
+            name: "projectName",
+            message: "Project name (use lowercase and dashes)",
+          },
+        ])
+      ).projectName;
+    }
+
+    const packageJsonTemplate = await fs.readFile(
+      this.templatePath("package.json"),
+      { encoding: "utf-8" }
+    );
+    const packageJsonStr = ejs.render(packageJsonTemplate, {
+      orgName: this.options.orgName,
+      projectName: this.options.projectName,
+    });
+
+    const packageJson = JSON.parse(packageJsonStr);
+
+    // if (this.options.typescript) {
+    //   // Will be added as a dependency via ts package.json
+    //   delete packageJson.devDependencies["@types/jest"];
+    // }
+
+    this.fs.extendJSON(this.destinationPath("package.json"), packageJson);
+
+    // if (this.options.typescript) {
+    //   this.fs.extendJSON(
+    //     this.destinationPath("package.json"),
+    //     this.fs.readJSON(
+    //       this.templatePath("../../common-templates/typescript/package.json")
+    //     )
+    //   );
+    // }
+  }
+  async copyOtherFiles() {
+    this.options.framework = "svelte";
+
+    const srcFileExtension = /* this.options.typescript ? "tsx" : */ "js";
+
+    this.fs.copyTpl(
+      this.templatePath("../../common-templates/gitignore"), // this is relative to /templates
+      this.destinationPath(".gitignore"),
+      this.options
+    );
+    // this.fs.copyTpl(
+    //   this.templatePath(".prettierignore"),
```

Comment:
Are you planning on adding prettier to the svelte generator? Does prettier know how to handle svelte files?

Reply:
Prettier is supported though a plugin https://github.com/sveltejs/prettier-plugin-svelte